File: Task 4/Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import android.os.Bundle;
import dev.ogabek.java.R;
import dev.ogabek.java.manager.AuthManager;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.iv_logout).setOnClickListener(view -> {
            AuthManager.signOut();
            callSignInActivity();
        });

    }
}

File: Task 3/Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.manager.AuthManager

class MainActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<ImageView>(R.id.iv_logout).setOnClickListener {
            AuthManager.signOut()
            callSignInActivity()
        }

    }
}